web/text_layer_builder.js

```diff
@@ -314,30 +319,37 @@ var TextLayerBuilder = (function TextLayerBuilderClosure() {
      */
     _bindMouse: function TextLayerBuilder_bindMouse() {
       var div = this.textLayerDiv;
+      var self = this;
       div.addEventListener('mousedown', function (e) {
+        if (self.enhanceTextSelection && self.textLayerRenderTask) {
+          self.textLayerRenderTask.expandTextDivs(true);
+        }
         var end = div.querySelector('.endOfContent');
         if (!end) {
           return;
         }
-//#if !(MOZCENTRAL || FIREFOX)
-        // On non-Firefox browsers, the selection will feel better if the height
-        // of the endOfContent div will be adjusted to start at mouse click
-        // location -- this will avoid flickering when selections moves up.
-        // However it does not work when selection started on empty space.
-        var adjustTop = e.target !== div;
-//#if GENERIC
-        adjustTop = adjustTop && window.getComputedStyle(end).
-          getPropertyValue('-moz-user-select') !== 'none';
-//#endif
-        if (adjustTop) {
-          var divBounds = div.getBoundingClientRect();
-          var r = Math.max(0, (e.pageY - divBounds.top) / divBounds.height);
-          end.style.top = (r * 100).toFixed(2) + '%';
-        }
-//#endif
-        end.classList.add('active');
-      });
+  //#if !(MOZCENTRAL || FIREFOX)
+          // On non-Firefox browsers, the selection will feel better if the
+          // height of the endOfContent div will be adjusted to start at mouse
+          // click location -- this will avoid flickering when selections moves
+          // up. However it does not work when selection started on empty space.
+          var adjustTop = e.target !== div;
+  //#if GENERIC
+          adjustTop = adjustTop && window.getComputedStyle(end).
+            getPropertyValue('-moz-user-select') !== 'none';
+  //#endif
+          if (adjustTop) {
+            var divBounds = div.getBoundingClientRect();
+            var r = Math.max(0, (e.pageY - divBounds.top) / divBounds.height);
+            end.style.top = (r * 100).toFixed(2) + '%';
+          }
+  //#endif
+          end.classList.add('active');
+          });
```

Comment:
Please revert all the indentation/line-breaking changes made above.
(Also note that the preprocessor tags, i.e. lines starting with `//#` cannot be indented, since that actually breaks the preprocessor.)